### PR TITLE
Build processor via makefile

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR adds support for building the processor via the `make dist` or `make processor` commands.  It also removes building the plugin for non-supported environments.

#### Ticket Link
NONE

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -74,13 +76,25 @@ ifneq ($(MM_SERVICESETTINGS_ENABLEDEVELOPER),)
...
-	cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-amd64;
-	cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-arm64;
-	cd server && env CGO_ENABLED=0 GOOS=windows GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-windows-amd64.exe;
+##  only AMD64 Linux is supported for now.
+##	cd server && env CGO_ENABLED=0 GOOS=linux GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-linux-arm64;
+##	cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-amd64;
+##	cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-arm64;
+##	cd server && env CGO_ENABLED=0 GOOS=windows GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-windows-amd64.exe;
```
Comment (mickmister) [inline Makefile:83]: I think this will make it so dev machines can't run this except for `linux-amd64`. Maybe handling this should be in a separate PR

Comment (wiggin77) [reply Makefile:83]: Dev machines will build via the MM_SERVICESETTINGS_ENABLEDEVELOPER which builds the current architecuture.

Comment (wiggin77) [reply Makefile:83]: If I don't comment these out here, the resulting tarball includes all the binaries and the plugin is 75MB

Comment (mickmister) [reply Makefile:83]: @wiggin77 In practice many people don't have that env var enabled. It's common to have the server configured with developer mode, but in this case the env of the specific terminal compiling the plugin needs to have this env var set. I think we should generally assume that this env var is not set

Comment (wiggin77) [reply Makefile:83]: @mickmister  What do you suggest?  We cannot have all these non-supported architectures in the release.

Comment (mickmister) [reply Makefile:83]: @wiggin77 We can add an env var like `OFFICIAL_PLUGIN_RELEASE: true` in this workflow: https://github.com/mattermost/delivery-platform/blob/2332ed4f90470fd09a10c7f0677b31763712e919/.github/workflows/release-mattermost-plugins.yaml#L141

Which in turn calls this `plugin-ci` workflow that runs `make dist` https://github.com/mattermost/actions/blob/main/plugin-ci/build/action.yaml

Then the plugin repo's `Makefile` can conditionally choose which architectures should be built based on the value of the `OFFICIAL_PLUGIN_RELEASE` env var. I think setting this env var in the first workflow makes sense, since we're absolutely sure a release is being built in that context.

Comment (wiggin77) [reply Makefile:83]: @mickmister that sounds great, but I need a build today.  I'm not concerned with Mac/Windows support, even on dev machines at this point.  They can set a env var locally as a workaround for now.

Comment (mickmister) [reply Makefile:83]: Alternatively, we can make it so `make dist` is only used by CI for this plugin or something like that. Either way, I think some assumption on the side of CI or the plugin needs to be changed as far as how to handle compiling the release

Comment (wiggin77) [reply Makefile]: @mickmister I have made the change only for CI.  CI will build only Linux AMD64.

File: Makefile [hunk 2]
```diff
@@ -69,18 +71,32 @@ ifneq ($(MM_DEBUG),)
...
+	ifneq ($(IS_CI),true)
+		cd server && env CGO_ENABLED=0 GOOS=linux GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-linux-arm64;
+		cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-amd64;
+		cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-arm64;
+		cd server && env CGO_ENABLED=0 GOOS=windows GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-windows-amd64.exe;
+	endif
 endif
 endif
 
+## Builds the processor for all supported architectures.
```
Comment (wiggin77) [inline Makefile:89]:
```suggestion
## Builds the processor for all architectures.
```

**Review by fmartingr — APPROVED** [review]
Approving this for urgency, but I would agree with the use of `MM_SERVICESETTINGS_ENABLEDEVELOPER` for local development (which I personally use so builds are faster). We should spread its usage if people are not using it.

**Review by mickmister — APPROVED** [review]
